test_ambaridb-onefs.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/sh
+
+# Usage info
+show_help() {
+cat << EOF
+Usage: ${0##*/} [-o ONEFSHOST]
+Execute on host with Ambari Server to correct the host and cluster state due to
+bug in OneFS 8.0.0.3 and 8.0.1.0 when used with Ambari 2.4.x.
+
+    -o ONEFSHOST   The OneFS host name as it appears in Ambari Server.
+EOF
+}
+
+ONEFSHOSTNAME=""
+
+OPTIND=1
+
+while getopts ho: opt; do
+    case $opt in
+        h)
+            show_help
+            exit 0
+            ;;
+        o)  ONEFSHOSTNAME=$OPTARG
+            ;;
+        *)
+            show_help >&2
+            exit 1
+            ;;
+    esac
+done
+
+echo "Beginning review and edits of Ambari database for OneFS host $ONEFSHOSTNAME."
+export PGPASSWORD=bigdata
+if [ -f /var/run/ambari-server/ambari-server.pid ]; then
+    AMBARION="true"
+else
+    AMBARION="false"
+fi
+ONEFSHOSTID=$(psql -At -U ambari -c "SELECT host_id from hosts WHERE host_name='$ONEFSHOSTNAME';") &&
```

Comment:
The `&&` is testing the assignment (always true afaik) here, but I think you mean it to test the command in the subshell (`$()`). I am not aware of a simple way to cease execution if that command fails :/ (The `&&` isn't hurting anything, though, just a comment FYI.)